lib/hub-registry.js

```diff
@@ -126,18 +103,9 @@ HubRegistry.prototype.init = function (taker) {
  * @param {function} fn - the task function
  */
 HubRegistry.prototype.set = function (name, fn) {
+	// TODO: what if this collides with a master task?
+	ForwardRegistry.prototype.set.call(this, name, fn);
```

Comment:
This is called because you are overwriting the `set` method which is called by `gulp.task`, so if you call gulp.task in the entry gulpfile, the tasks are actually registered without this call.  Maybe there is some different logic to add, but this got the example to work and the tests to pass.